src/ripple/basics/impl/ProgressDisplay.cpp

```diff
@@ -0,0 +1,105 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2020 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <ripple/basics/ProgressDisplay.h>
+#include <iomanip>
+#include <iostream>
+
+namespace ripple {
+
+ProgressDisplay::ProgressDisplay(
+    std::uint64_t const expectedCount,
+    std::string const& countedItemName,
+    std::string const& initialMessage)
+    : expectedCount_(expectedCount)
+    , countedItemName_(countedItemName)
+    , initialMessage_(initialMessage)
+{
+    updateDisplay(0);
+}
+
+ProgressDisplay::~ProgressDisplay()
+{
+    if (itemsCompleted_ < expectedCount_)
+    {
+        std::cout << "Incomplete!" << std::endl;
+        std::cout.flush();
+    }
+}
+
+ProgressDisplay&
+ProgressDisplay::operator++()
+{
+    if (itemsCompleted_ < expectedCount_)
+    {
+        // Ignore calls to this function when the
+        // expectedCount_ has been reached
+
+        ++itemsCompleted_;
+
+        if (auto const percent = (100.0 * itemsCompleted_) / expectedCount_;
+            percent >= progressUpdateThresh_)
+        {
+            updateDisplay(percent);
+            ++progressUpdateThresh_;
+        }
+    }
+
+    return *this;
+}
+
+void
+ProgressDisplay::updateDisplay(double const percent)
+{
+    if (!itemsCompleted_)
+        std::cout << std::endl
+                  << std::string(initialMessage_.size(), '=') << std::endl
+                  << initialMessage_ << std::endl
+                  << std::string(initialMessage_.size(), '=') << std::endl;
```

Comment:
I wonder if it makes more sense to move this output to the body of the constructor and unconditionalized it...

Reply:
Done